kernel/api/src/main/java/org/sakaiproject/time/api/UserTimeService.java

```diff
@@ -34,17 +34,89 @@
      * @return true if successful
      */
     boolean clearLocalTimeZone(String userId);
-    
+
+    /**
+     * Gets the time formatter with the given formatting style in the user's locale and preferred timezone.
+     * @param date
+     * @param locale
+     * @return
+     */
+    public String  timeFormatShort(Date date, Locale locale);
```

Comment:
Does it make sense to add another param that depicts the requested format of
{short, medium, long}
This prevents api from increasing every time another option is needed.
Have a look at the enum java.time.temporal.ChronoUnit to get an idea of how this could be done.

essentially you would have:
```
enum FormatUnit {
	SHORT("Short", 0),
	MEDIUM("Medium", 1),
	LONG("Long", 2),
}
```
This would allow for simpler concise methods:
`public String timeFormat(Date date, Locale locale, FormatUnit format);`
`public String dateFormat(Date date, Locale locale, FormatUnit format);`
`public String dateTimeFormat(Date date, Locale locale, FormatUnit format);`

Reply:
For sure it's cleaner to have a single method, but then you need to change lots of tool code buried in JSPs and do lots of regression testing. I added new methods to avoid massive tool changes.

Reply:
If you want to keep the existing methods working (to avoid massive tool changes) then forward the older methods to the newer ones and mark them as deprecated. Also if you worried about regressions unit tests will eliminate those concerns as well.

Reply:
Tool changes are way less than I thought. Will update this PR shortly with changes.